### ISSUE TITLE
riak_repl2_rtsource_conn invalid state in 2.0.5 [JIRA: RIAK-1669]

### ISSUE THREAD

**Issue by ian-mi** [basho/riak_repl#666]
In Riak 2.0.5, riak_repl2_rtsource_conn crashes with function_clause on a handle_call because its state is replaced by {noreply, state}. This extra noreply seems to originate from the reconnect function in the newly added rebalance code.

**Comment by ian-mi** [comment]
An example crash report:

```
2015-04-02 01:23:16.223 [error] <0.5102.0> CRASH REPORT Process <0.5102.0> with 0 neighbours exited with reason: no function clause matching riak_repl2_rtsource_conn:terminate({function_clause,[{riak_repl2_rtsource_conn,handle_call,[{connected,#Port<0.32877>,ranch_tcp,{...},...},...],...},...]},{noreply,{state,"clustername",undefined,#Ref<0.0.0.251966>,undefined,undefined,undefined,undefined,...}}) line 329 in gen_server:terminate/6 line 722
```

**Comment by bsparrow435** [comment]
Working on a riak_test for this now.

**Comment by bsparrow435** [comment]
Never linked the test to the issue haha: https://github.com/basho/riak_test/pull/811

**Comment by Basho-JIRA** [comment]
Doug confirmed this has been fixed and merged in both 2.0 and 2.1 

**_[posted via JIRA by Patricia Brewer]**_

**Comment by Basho-JIRA** [comment]
Fixed and merged into 2.0 and 2.1 per Doug. 

**_[posted via JIRA by Patricia Brewer]**_
